### PR TITLE
Feature/3020/creating seeder

### PR DESCRIPTION
**Pull request by MayankBisht8630**
Targets #3020

### Description
Creating Seeder For CodeTrek

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: database/Seeders/CodetrekSeeder.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            'last_name'=>$faker->email,
+            'email'=>$faker->email,
+            'github_user_name'=>'test',
+            'phone'=>'',
+            'course'=>'',
+            'start_date'=>'',
+            'graduation_year'=>'',
+            'university'=>'',
+            'created_at'=>'',
+            'updated_at'=>'',
```
Comment (GauravGusain98) [inline database/Seeders/CodetrekSeeder.php:32]: These 2 columns will be automatically updated. No need to pass then in insert statement.

File: database/Seeders/CodetrekSeeder.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        for ($i = 1; $i <= 5; $i++) {
+            DB::table('code_trek_applicants')->insert([
+            'first_name'=>$faker->name,
+            'last_name'=>$faker->email,
+            'email'=>$faker->email,
+            'github_user_name'=>'test',
+            'phone'=>'',
+            'course'=>'',
+            'start_date'=>'',
+            'graduation_year'=>'',
```
Comment (GauravGusain98) [inline database/Seeders/CodetrekSeeder.php:29]: Fill the vlaue in these fields as well using faker

File: database/Seeders/CodetrekSeeder.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            'university'=>'',
+            'created_at'=>'',
+            'updated_at'=>'',
+            'deleted_at'=>'',
```
Comment (GauravGusain98) [inline database/Seeders/CodetrekSeeder.php:33]: remove this as well

File: database/Seeders/CodetrekSeeder.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            'round_name'=>'level 1'
```
Comment (GauravGusain98) [inline database/Seeders/CodetrekSeeder.php:34]: take value from config file. Don't hardcode the value

File: database/Seeders/CodetrekSeeder.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        $faker = Faker::create();
+
+        for ($i = 1; $i <= 5; $i++) {
```
Comment (GauravGusain98) [inline database/Seeders/CodetrekSeeder.php:20]: For this you need to create new seeder file. The `CodetrekSeeder` file is only used to call those seeder files. 

Look at how projectSeeder is working.

File: database/Seeders/CodetrekSeeder.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            DB::table('code_trek_applicants')->insert([
```
Comment (GauravGusain98) [inline database/Seeders/CodetrekSeeder.php:21]: When we create a new applicant we also create entry in the `codetrek_applicant_round` table. So for every applicant you also need to create entry in the `codetrek_applicant_round` table

File: config/database.php [hunk 7]
```diff
@@ -47,8 +47,8 @@
             'username' => env('DB_USERNAME', 'forge'),
             'password' => env('DB_PASSWORD', ''),
             'unix_socket' => env('DB_SOCKET', ''),
-            'charset' => 'utf8mb4',
-            'collation' => 'utf8mb4_unicode_ci',
+            'charset' => 'utf8',
+            'collation' => 'utf8_general_ci',
```
Comment (GauravGusain98) [inline config/database.php:51]: These changes are not needed. Please revert them. Also please do a self review of your code changes before assigning PR.

File: database/Seeders/DatabaseSeeder.php [hunk 8]
```diff
@@ -23,5 +23,6 @@ public function run()
         // books
         $this->call(BooksPermissionsSeeder::class);
         $this->call(BookCategoriesTableSeeder::class);
+        $this->call(CodetrekSeeder::class);
```
Comment (GauravGusain98) [inline database/Seeders/DatabaseSeeder.php:26]: This is not needed.

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comments

File: Modules/CodeTrek/Database/factories/CodetrekApplicantRoundFactory.php [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+/** @var \Illuminate\Database\Eloquent\Factory $factory */
+
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Modules\CodeTrek\Entities\CodeTrekApplicant;
+use Modules\CodeTrek\Entities\CodeTrekApplicantRoundDetail;
+
+class CodeTrekApplicantFactory extends Factory
```
Comment (GauravGusain98) [inline Modules/CodeTrek/Database/factories/CodetrekApplicantRoundFactory.php:9]: The name of this Class is not correct.

**Comment by Ayush8923** [comment]
@MayankBisht8630 Please resolve the merge conflicts.

**Comment by MayankBisht8630** [comment]
Ok sir

**Review by Ayush8923 — CHANGES_REQUESTED** [review]

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left some comments. 
And Have you tested it locally?

**Comment by MayankBisht8630** [comment]
> Left some comments. And Have you tested it locally?

Yes Sir made required Changes , And I have tested it locally working fine

File: Modules/CodeTrek/Database/Factories/CodeTrekApplicantFactory.php [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            'last_name'=>$faker->lastName,
+            'email'=>$faker->email,
+            'github_user_name'=>$faker->userName,
+            'phone'=>$faker->phoneNumber,
+            'course'=>$faker->word,
+            'start_date' => Carbon::today(),
+            'university' =>$this->getCollegeNames()[array_rand($this->getCollegeNames())],
+            'graduation_year'=>Carbon::now()->year,
+        ];
+    }
```
[the remark this answers is not in the document]
Comment (MayankBisht8630) [reply Modules/CodeTrek/Database/Factories/CodeTrekApplicantFactory.php:33]: > 
Sir If I am Adding a line after 33 so it is showing error in checks

**Review by Ayush8923 — APPROVED** [review]

**Comment by Ayush8923** [comment]
@GauravGusain98  Please take a look. You have also placed a change request in this PR.

**Pull request merged**